ext/web/13_message_port.js

```diff
@@ -85,16 +86,37 @@ const MessageChannelPrototype = MessageChannel.prototype;
 const _id = Symbol("id");
 const MessagePortIdSymbol = _id;
 const _enabled = Symbol("enabled");
+const _nodeWorker = Symbol("nodeWorker");
+const _closeEventDispatched = Symbol("closeEventDispatched");
 
 /**
  * @param {number} id
  * @returns {MessagePort}
  */
-function createMessagePort(id) {
+function createMessagePort(id, nodeWorker = false) {
+  if (nodeWorker) {
+    defineEventHandler(MessagePort.prototype, "close");
+  }
   const port = webidl.createBranded(MessagePort);
   port[core.hostObjectBrand] = core.hostObjectBrand;
   setEventTargetData(port);
   port[_id] = id;
+  port[_nodeWorker] = nodeWorker;
+  if (nodeWorker) {
+    port.on = port.addListener = function (name, listener) {
+      const _listener = (ev) => listener(ev.data);
+      if (name == "message") {
+        port.onmessage = _listener;
+      } else if (name == "messageerror") {
+        port.onmessageerror = _listener;
+      } else if (name == "close") {
+        port.onclose = _listener;
+      } else {
+        throw new Error("Unsupported event: " + name);
+      }
+      return port;
+    };
+  }
```

Comment:
I think all of this can be done in `node:worker_threads` similarly to what we do for `parentPort`

Reply:
Ah, I'll try that. I think I had to use a flag in web implementation because deserialization of workerData didn't setup the MessagePort class correctly in worker thread.